### ISSUE TITLE
Issue using RealSenseFrameGrabber

### ISSUE THREAD

**Issue by mattdibi** [bytedeco/javacv#669]
I encountered some problems using the RealSenseFrameGrabber class.
I have managed to display the color stream but I cannot access the depth stream which I need for my application.

in my Main i have the following:

```java
...
35 RealSenseFrameGrabber grabber = RealSenseFrameGrabber.createDefault(0);
36 grabber.tryLoad();
37 grabber.enableDepthStream();
38 grabber.start();
...
```

when I try to run the code I get:

```sh
RealSense devices found: 1
Exception in thread "main" java.lang.NullPointerException
	at org.bytedeco.javacv.RealSenseFrameGrabber.enableDepthStream(RealSenseFrameGrabber.java:156)
	at Main.main(Main.java:37)
```
RealSenseFrameGrabber.java:156
```java
device.enable_stream(RealSense.depth, depthImageWidth, depthImageHeight, RealSense.z16, depthFrameRate)
```
because nobody loaded the device variable...
 How am I supposed to use this class?

**Comment by saudet** [comment]
@poqudrof Is this a bug?

**Comment by poqudrof** [comment]
Sorry for the late answer.
The ```device```  is loaded with the  ``` tryLoad() ``` call. You can call it once before instanciating objects. The ```tryLoad``` is a **static** method. 

```
...
35 RealSenseFrameGrabber.tryLoad();
36 RealSenseFrameGrabber grabber = RealSenseFrameGrabber.createDefault(0);
37 grabber.enableDepthStream();
38 grabber.start();
...
```

However, I do no think that it is what causes the issue. 

I have a few questions to help you out better: 
- Which camera do you use ? I tested it mostly on ZR300, and tried it on R200.  
- What is your OS / linux distro ?
- Is the depth working with the realsense binary? Here is why:

The bindings are a little out of date, as they work on [1.9.6](https://github.com/IntelRealSense/librealsense/releases/tag/v1.9.6). 

You must succeed to compile, install and run the ```./bin/cpp-config-ui``` and see the depth stream properly. If you don't see the depth there, you may need to patch your linux UVC drivers. To do so have a look at the [installation guides of the librealsense](https://github.com/IntelRealSense/librealsense#installation-guide). 

The dependency check and kernel patching is quite cumbersome but the realsense guys push for it to be standard in the next linux kernels. 

Cheers, 
Jeremy.

**Comment by mattdibi** [comment]
Ok I just tested it with this code:

```java
RealSenseFrameGrabber.tryLoad();                                                                                                                                                                                              
RealSenseFrameGrabber grabber = RealSenseFrameGrabber.createDefault(0);
grabber.enableDepthStream();
grabber.enableColorStream();
grabber.start();
```



 and it gives me the usual:
```
RealSense devices found: 1
java.lang.NullPointerException
```
upon using the enableStream function.

About your questions:

- I use a SR300 and a R200
- I'm working on a Linux Mint 18.1 and targeting a custom linux distribution developed with Yocto project
- Yes, everything is working fine with binaries and my C++ version of the code on both my systems

Eventually I needed to write my own code to use the RealSense cameras in Java. You can take a look at it [here](https://github.com/mattdibi/RSPassengerCounter/tree/master/java_src)

**Comment by poqudrof** [comment]
It seems quite weird the `NullPointerException` you encounter. Can you get a stackstrace to indentify the exact line ?

``` java 
try { 

} Catch(Exception e){
 e.printStackTrace 
}
```

**Comment by mattdibi** [comment]
Here you go:

```
RealSense devices found: 1
java.lang.NullPointerException
	at org.bytedeco.javacv.RealSenseFrameGrabber.enableDepthStream(RealSenseFrameGrabber.java:156)
	at Main.main(Main.java:17)
```
Line 17 is `grabber.enableDepthStream();`
Tried with both my cameras (R200 and SR300)

**Comment by poqudrof** [comment]
Ahhhh, I see the issue. I did not see the latest changes in the repo, it also broke my code I remember now. 

I will sum up the issue: 
- Changes occured in the [previous commit](https://github.com/bytedeco/javacv/commit/8b566f17225965ee91b9d30e6db49bd4f3c91095). 
- Before the `device` was created when the FrameGrabber was created (line 121)
- After the `device` is created at the moment when the camera is started. 
My guess was, this initialization was problematic for external programs that use JavaCV like ProCamCalib and that's why @saudet moved it away. 

The problem is that it does not behave like one camera but like a device that can contain multiple cameras. 

Actually I just out I corrected the issue on my computer a while ago and did not commit nor published it... My bad. I test it out again and create a pull request with it.
